modules/yandexAnalyticsAdapter.js

```diff
@@ -0,0 +1,222 @@
+import buildAdapter from '../libraries/analyticsAdapter/AnalyticsAdapter.js';
+import adapterManager from '../src/adapterManager.js';
+import { logError, logInfo } from '../src/utils.js';
+import CONSTANTS from '../src/constants.json';
+import * as events from '../src/events.js';
+
+const tagTlds = ['com', 'ru'];
+const fileNames = ['tag', 'int']
+const tagUrlTemplate = 'https://mc.yandex.{tld}/metrika/{file}.js';
+export const tagURLs = tagTlds.flatMap((tld) => {
+  const partialTemplate = tagUrlTemplate.replace('{tld}', tld);
+  return fileNames.map((file) => partialTemplate.replace('{file}', file));
+});
+
+const timeoutIds = {};
+const tryUntil = (operationId, conditionCb, cb) => {
+  if (!conditionCb()) {
+    cb();
+    timeoutIds[operationId] = setTimeout(
+      () => tryUntil(conditionCb, conditionCb, cb),
+      100
+    );
+  }
+};
+const waitFor = (operationId, conditionCb, cb) => {
+  if (conditionCb()) {
+    cb();
+  } else {
+    timeoutIds[operationId] = setTimeout(() => waitFor(conditionCb, cb), 100);
+  }
+};
+
+const clearWaitForTimeouts = (timeouts) => {
+  timeouts.forEach((timeoutID) => {
+    if (timeoutIds[timeoutID]) {
+      clearTimeout(timeoutIds[timeoutID]);
+    }
+  });
+};
+
+const SEND_EVENTS_BUNDLE_TIMEOUT = 1500;
+const {
+  EVENTS: {
+    BID_REQUESTED,
+    BID_RESPONSE,
+    BID_ADJUSTMENT,
+    BID_WON,
+    BIDDER_DONE,
+    AUCTION_END,
+    BID_TIMEOUT,
+  },
+} = CONSTANTS;
+
+export const EVENTS_TO_TRACK = [
+  BID_REQUESTED,
+  BID_RESPONSE,
+  BID_ADJUSTMENT,
+  BID_WON,
+  BIDDER_DONE,
+  AUCTION_END,
+  BID_TIMEOUT,
```

Comment:
Wouldn't these all be undefined since they're declared inside the EVENTS object above?

Reply:
There would be ["bidRequested", "bidResponse", "bidAdjustment", ...]. lines 42-52 have [destruction syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment), so for example BID_REQUESTED will be CONSTANTS.EVENTS.BID_REQUESTED, that is "bidRequested" as [defined here](https://github.com/prebid/Prebid.js/blob/44b7c1591616a2a9d01b96fc8640be113ac92eb6/src/constants.json#L30) and so on